kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/SubjectAccessReviewDSLImpl.java

```diff
@@ -0,0 +1,154 @@
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl.internal;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.authorization.*;
+import io.fabric8.kubernetes.api.model.authorization.DoneableLocalSubjectAccessReview;
+import io.fabric8.kubernetes.api.model.authorization.DoneableSubjectAccessReview;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.dsl.Createable;
+import io.fabric8.kubernetes.client.dsl.SubjectAccessReviewDSL;
+import io.fabric8.kubernetes.client.dsl.base.OperationContext;
+import io.fabric8.kubernetes.client.dsl.base.OperationSupport;
+import io.fabric8.kubernetes.client.utils.Utils;
+import okhttp3.OkHttpClient;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class SubjectAccessReviewDSLImpl extends OperationSupport implements SubjectAccessReviewDSL {
+
+  private static final String AUTHORIZATION = "authorization.k8s.io/v1";
+  private static final String SAR_PLURAL = "subjectaccessreviews";
+  private static final String LSAR_PLURAL = "localsubjectaccessreviews";
+
+  private boolean isNamespaced = false;
+
+  public SubjectAccessReviewDSLImpl(OperationContext context) {
+    super(context.withApiGroupName(AUTHORIZATION).withPlural(SAR_PLURAL));
+  }
+
+  private SubjectAccessReviewDSLImpl(OperationContext context, String namespace, String plural, boolean isNamespaced) {
+    super(context.withApiGroupName(AUTHORIZATION).withNamespace(namespace).withPlural(plural));
+    this.isNamespaced = isNamespaced;
+  }
+
+  public SubjectAccessReviewDSLImpl(OkHttpClient client, Config config) {
+    this(new OperationContext().withOkhttpClient(client).withConfig(config));
+  }
+
+  @Override
+  public boolean isResourceNamespaced() {
+    return isNamespaced;
+  }
+
+  @Override
+  public SubjectAccessReview create(SubjectAccessReview... resources) {
+    return new CreatableSubjectAccessReview().create(resources);
+  }
+
+  @Override
+  public io.fabric8.kubernetes.api.model.authorization.DoneableSubjectAccessReview createNew() {
+    return new CreatableSubjectAccessReview().createNew();
+  }
+
+  @Override
+  public Createable<LocalSubjectAccessReview, LocalSubjectAccessReview, DoneableLocalSubjectAccessReview> inNamespace(String namespace) {
+    //Should we use this namespace to set the metadata namespace and the spec namespace of the LocalSubjectAccessReview object?
+    return new SubjectAccessReviewDSLImpl(this.context, namespace, LSAR_PLURAL, true).local(namespace);
```

Comment:
In my opinion, this namespace should be used for setting metadata namespace only.

Reply:
That would mean the user has to provide the namepsace again in the spec, in addition to the `SubjectAccessReviewDSL`  when using a `LocalSubjectAccessReview`. This is redundant, don't you think so?

Reply:
Yeah, but this aligns with standard kubernetes flow. I think in kubernetes also it's done like this. Could you please check?

Reply:
Yes. I checked this. You are required to give the same namespace in the metadata and the spec if you are writing a yaml. But kubectl client has an easier syntax
```
kubectl auth can-i create deployments --namespace dev
```
This looks like a cleaner way to do this. I thought setting the namespace if the user has not set it, will make it easier to use. Thought?

Reply:
Hm, well we can then make an adjustment in its *OperationsImpl class that overrides default implementation.

Reply:
I have done it n `setNamespace()` method in the same class.